src/stage/supervisor.rs

```diff
@@ -71,51 +67,51 @@ impl SupervisorBuilder {
 }
 
 pub struct Supervisor {
-    session_id: usize,
-    reporters: Reporters,
+    session: usize,
     reconnect_requests: u8,
-    tx: Sender,
-    rx: Receiver,
     connected: bool,
     shutting_down: bool,
-    address: Address,
-    shard: StageNum,
-    reporters_num: u8,
+    address: String,
+    shards: u8,
+    tx: Sender,
+    rx: Receiver,
     supervisor_tx: node::supervisor::Sender,
+    reporters: Reporters,
 }
 
 impl Supervisor {
     pub async fn run(mut self) {
+        let reporters_num = self.reporters.capacity() as u8;
         // Create sender's channel
         let (sender_tx, sender_rx) = mpsc::unbounded_channel::<sender::Event>();
         // Prepare range to later create stream_ids vector per reporter
-        let (mut start_range, appends_num): (StreamId, StreamId) =
-            (0, 32767 / (self.reporters_num as i16));
+        let (mut start_range, appends_num): (Stream, Stream) =
+            (0, 32767 / (reporters_num as i16));
         // Start reporters
-        for reporter_num in 0..self.reporters_num {
+        for reporter_num in 0..reporters_num {
             // Create reporter's channel
             let (reporter_tx, reporter_rx) = mpsc::unbounded_channel::<reporter::Event>();
             // Add reporter to reporters map
             self.reporters.insert(reporter_num, reporter_tx.clone());
             // Start reporter
             let last_range = start_range + appends_num;
-            let stream_ids: StreamIds = ((if reporter_num == 0 {
+            let streams: Streams = ((if reporter_num == 0 {
                 1 // we force first reporter_num to start range from 1, as we reversing stream_id=0 for future uses.
             } else {
                 start_range // otherwise we keep the start_range as it's
             })..last_range)
                 .collect();
             start_range = last_range;
             let reporter_builder = reporter::ReporterBuilder::new()
-                .reporter_num(reporter_num)
-                .session_id(self.session_id)
-                .sender_tx(sender_tx.clone())
-                .supervisor_tx(self.tx.clone())
-                .stream_ids(stream_ids)
+                .reporter(reporter_num)
+                .session(self.session)
+                .streams(streams)
+                .shard(self.shards.clone())
+                .address(self.address.clone())
                 .tx(reporter_tx)
                 .rx(reporter_rx)
-                .shard(self.shard.clone())
-                .address(self.address.clone());
+                .supervisor_tx(self.tx.clone())
```

Comment:
For the naming rules I prefer to use node_tx here for ease of reading (and precise identification), so as other supervisor_tx field in other structs (reporter/stage/node/etc.) How do you think?